### ISSUE TITLE
Further explanations

### ISSUE THREAD

**Issue by y-nk** [upstash/vercel-note-taker-slackbot#1]
Thanks for this, both the post on Upstash blog and this repo look great. There's a bug regarding slash command which isn't covered and I was wondering if you could give your input.

The bug is that all slash command processing must happen below 3s, otherwise you need to ack right away and send a message _later on_. The problem being that, in Vercel when you do `res.status(200).end()` the worker thread gets destroyed and you can't proceed to that "later on" part.

Did you ever try? What's your approach there?

**Comment by burak-upstash** [comment]
Good catch! Since the sample is a fairly straight forward one, I simply neglected the tiny probability that the requests wouldn't be acknowledged in under 3 seconds.

I haven't tested your suggested solution but I am sure that there should be some sort of functionality either in express/nodejs or Vercel side that enables you to continue your process after sending a response.

For example, I see that res.write() functionality does not end the response process. 
Another approach might be acknowledging and then processing the command in an other function that can be called.

**Comment by burak-upstash** [comment]
If you have any questions or suggestions, let us now :) Closing the issue.

**Comment by y-nk** [comment]
@burak-upstash sorry for not following up earlier (and yes of course we can close this issue :) !)

To give some answer and maybe if someone is interested in this later on:

1. I tried `res.write` but this only writes in the buffer (does not flush it) so it's no use.

2. I also tried acknowledging "by default" and "chain" the processing to another request, but it wont work. If you do some kind of `fetch()` without `await` (means you don't wait for the response of the "child" request) then the lambda dies too soon and the request isn't sent at all ; else if you wait with `await`, then the full processing time of the child request is included inside the parent one since you await for response and then it's back to square one.

3. Lastly what I tried was to think "in the middle" by building a low level equivalent of `fetch` which would send the data to the child request and close the tcp connexion right after (so the data is fully sent but we don't wait for the answer) but it didn't work since Vercel (or rather lambdas, i guess) are made to terminate immediately when/if the socket connexion is lost.

Therefore I think it's not possible without the use of a third party which has different behavior than the lambdas, using it as a a pivot point. The easiest is probably a Github workflow since it is made to be fully async, but then there would be some billing involved at some point :)

When I think about it, it's most likely that lambda were designed not to be chainable at all, and it would make sense somehow.
